test/solvers/advanced/forward_backward_test.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test for the forward-backward solver."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import pytest
+import numpy as np
+
+# Internal
+import odl
+from odl.solvers import forward_backward_pd
+from odl.util.testutils import all_almost_equal, example_element
+
+# Places for the accepted error when comparing results
+PLACES = 8
+
+
+def test_forward_backward_inpit_handeling():
+    """Test to see that input is handled correctly.
+    """
+
+    # Create spaces
+    space1 = odl.uniform_discr(0, 1, 10)
+
+    # Create "correct" set of operators
```

Comment:
Comments in tests should be used very sparely. The only two use cases are (1) to structure tests (e.g. one section for real, one for complex) and (2) explain complex tests or non-obvious things that happen.

In this case, the comment below on the expected behavior is good, the others can be deleted without loss of information.